api/parts/data/fetch.js

```diff
@@ -2077,6 +2078,40 @@ fetch.jobDetails = async function(metric, params) {
     common.returnOutput(params, { sEcho: params.qstring.sEcho, iTotalRecords: total, iTotalDisplayRecords: total, aaData: items || [] });
 };
 
+/**
+ * Suspend a job from UI
+ * @param {string} metric - name of the collection where to get data from
+ * @param {object} params - params object with job id
+ * @returns {object} - result of the operation
+ **/
+fetch.suspendJob = async function(metric, params) {
+    try {
+        let jobStatus = params.qstring.suspend ? 7 : 0;
+        const result = await common.db.collection('jobs').findOneAndUpdate(
```

Comment:
This will leave already running / cancelled / whatever jobs in inconsistent state. findOneAndUpdate must only be used with positive operators like $eq

Reply:
And could you move logic to handle.js? :D